src/nvim/option.c

```diff
@@ -7111,16 +7111,26 @@ int csh_like_shell(void)
   return strstr((char *)path_tail(p_sh), "csh") != NULL;
 }
 
-/// Return true when window "wp" has a column to draw signs in.
-bool signcolumn_on(win_T *wp)
+/// Return the number of requested sign columns, based on current
+/// buffer signs and on user configuration.
+int win_signcol_count(win_T *wp)
 {
     if (*wp->w_p_scl == 'n') {
-      return false;
+      return 0;
     }
     if (*wp->w_p_scl == 'y') {
-      return true;
+      return 1;
+    }
+    if (*wp->w_p_scl == '2') {
```

Comment:
Could just as well allow the entire range `'1'` to `'9'` and calculate the value programatically.

Reply:
Agreed

Reply:
In fact, this requires more changes - `p_scl_values` needs to be extended as well.